### ISSUE TITLE
3.5.0 network request hangs pending. rolling back to 3.4.17 fixes an issue. 3.5.9 issue persists,

### ISSUE THREAD

**Issue by JustFly1984** [apollographql/apollo-client#9456]
**Intended outcome:**
correct network request execution

**Actual outcome:**
request hangs in pending state.

**How to reproduce the issue:**
I'm not sure how to reproduce. We had cached this error while updating @apollo/client@3.4.16 to 3.5.9. I've tested other versions and the bug appears in 3.5.0 and consists in every version till current at the time of writing 3.5.9.

We have nextjs@12.1 app, and we are requesting graphql on server-side with React.js hook, generated by @graphql-codegen/cl@2.6.2

**Versions**
 ```sh
 System:
    OS: macOS 11.6.4
  Binaries:
    Node: 14.15.4 - ~/.nvm/versions/node/v14.15.4/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 7.12.1 - ~/.nvm/versions/node/v14.15.4/bin/npm
  Browsers:
    Chrome: 98.0.4758.109
    Firefox: 91.0.1
    Safari: 15.3
  npmPackages:
    @apollo/client: 3.4.17 => 3.4.17  does work. 3.5.0+ doesn't
    @apollo/react-hooks: 4.0.0 => 4.0.0 
    @apollo/react-testing: 4.0.0 => 4.0.0 
```

Thank you for a great work!

**Comment by damiangreen** [comment]
I'm also getting this.. is there any solution?

**Comment by iprignano** [comment]
Are you by any chance using `getDataFromTree` and lazy queries (`useLazyQuery`) across your application?

**Comment by damiangreen** [comment]
Yes we are using       `await getDataFromTree(rootComponent)` and useLazyQuery

**Comment by iprignano** [comment]
@damiangreen You wouldn't have to remove `getDataFromTree` – but set the `ssr` option in the lazy query call to `false`, e.g.

```ts
useLazyQuery(query, {
  ssr: false,
  ...otherOptions
})
```

**Comment by damiangreen** [comment]
thx ill give it a go

**Comment by andreialecu** [comment]
Also running into queries randomly hanging when using `apollo-angular` so this is not a React only issue. 

Noticed this on `3.6.3`. It appears that things work for a while then suddenly a certain query will always hang. The promise will never return. I'm not noticing the query being issued into the network tab of Chrome Dev Tools at all.

It seems that downgrading to `3.4.17` fixes it indeed.

This seems of critical priority.

**Comment by zuspan** [comment]
I'm still having this issue with queries hanging on pending. The request is made, the payload sent, the server returns the response, but the browser still shows it as pending.

@apollo/client 3.6.9
node 16.16.0

is anyone else still seeing this or have a solution?

**Comment by JustFly1984** [comment]
Fixed in 3.6
